.github/workflows/qubes-dom0-packagev2.yml

```diff
@@ -0,0 +1,117 @@
+name: Test build and package QubesOS RPMs
+
+on:
+  workflow_call:
+    inputs:
+      qubes-component:
+        description: >
+          Name of QubesOS component as recognized by its build system.
+        required: true
+        type: string
+
+jobs:
+  build-and-package:
+    runs-on: ubuntu-latest
+    name: Compile and package RPM
+    permissions:
+      # for publishing releases
+      contents: write
+
+    steps:
+      - name: Install dependencies of builder script
+        # README also specified: python3-yaml rpm tree gpg openssl python3-setuptools
+        # docker.io was changed to docker because of conflict on containerd
+        run: |
+          sudo apt install python3-packaging createrepo-c devscripts \
+          docker python3-docker reprepro python3-pathspec \
+          mktorrent python3-lxml python3-dateutil
+
+      - uses: actions/checkout@v3
+        with:
+          repository: QubesOS/qubes-builderv2
+
+      - name: Cache Docker image and dom0 stuff
+        uses: actions/cache@v3
+        id: docker-cache
+        with:
+          path: |
+            /tmp/qubes-builder-fedora.tar
+            /tmp/cache/dom0.tar
+          key: |
+            ${{ hashFiles('tools/*') }}-docker-container
+
+      - name: Load Docker image
+        if: steps.docker-cache.outputs.cache-hit == 'true'
+        run: |
+          docker load --input /tmp/qubes-builder-fedora.tar
+
+      - name: Build Docker image (optional)
+        if: steps.docker-cache.outputs.cache-hit != 'true'
+        run: |
+          sed -i "s/RUN useradd -m user$/RUN useradd -m user -u $UID/" dockerfiles/fedora.Dockerfile
+          tools/generate-container-image.sh docker
+
+      - name: Export Docker image (optional)
+        if: steps.docker-cache.outputs.cache-hit != 'true'
+        run: |
+          docker save --output /tmp/qubes-builder-fedora.tar \
+                      qubes-builder-fedora:latest
+
+      - name: Prepare dom0 cache storage (optional)
+        if: steps.docker-cache.outputs.cache-hit != 'true'
+        run: |
+          mkdir --mode=777 /tmp/cache
+
+      - name: Prepare configuration
+        env:
+          URL: ${{ github.repositoryUrl }}
+          branch: ${{ github.head_ref }}
+          tag: ${{ github.ref_name }}
```

Comment:
These two variables can be removed unless you want to start using them below.

Reply:
Forgot to add the comment?  It would be helpful as I starred directly at the line and didn't see double expansion.

Reply:
Right, added now in https://github.com/TrenchBoot/.github/compare/fff3d731a201e585640efb0b42b0a88991f65465..c97347a77cd0575cc2ab8212ecd4a54dd6dc1593, thanks for catching it.

Reply:
@krystian-hebel Coming back to these lines.  `github.head_ref` seems to be specific to PRs (from [here](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context)):

> The `head_ref` or source branch of the pull request in a workflow run. 

And is empty on pushes.

`github.ref` seems to be available in more cases, have you tried using it and it didn't work for some reason or you just didn't try it?

Reply:
I tried it, it worked for one of those but not the other, don't remember which one worked. For the other one it returned _something_ but it was neither commit digest nor branch name, and as such it couldn't be checked out. I don't recall the details, but IIRC it was something that looked like the way Gerrit does it, with patchset numbers, but I may be wrong. Qubes builder didn't like it so I didn't play with it too much.